db.py

```diff
@@ -38,6 +38,9 @@ def record_would_play(self, player, game):
         self._store.add(wp)
         self._prune_expired()
         return wp
+    
+    def cancel_would_plays(self, player):
+        self._store = filter(lambda x:x.player != player, self._store)
```

Comment:
Does `filter()` return a set when called on a set? I didn't think it did but if it does then this is fine.